CMakeLists.txt

```diff
@@ -105,12 +107,50 @@ else()
   # find_package looks for already-installed system packages.
   # Configure with `-DCMAKE_PREFIX_PATH="/path/to/package1;/path/to/package2"`
   # to add search paths.
+  if (ANDROID)
+    message(STATUS "Please place the version-appropriate SDL3.aar in android-project/app/libs")
+  endif()
   find_package(SDL3 CONFIG REQUIRED)
 
   find_package(iniparser REQUIRED CONFIG COMPONENTS shared)
   target_link_libraries(Isle::iniparser INTERFACE iniparser-shared)
 endif()
 
+# Effectively use as a gate for running the Android build
+if (ANDROID_NDK_PATH)
+
+  FetchContent_GetProperties(SDL3)
+
+  set(SDL_ANDROID_ARCHIVE "${CMAKE_SOURCE_DIR}/android-project/app/libs/SDL3.aar")
+  add_custom_command(OUTPUT "${SDL_ANDROID_ARCHIVE}"
+    COMMAND ${Python3_EXECUTABLE} "${SDL3_SOURCE_DIR}/build-scripts/build-release.py"
+       --actions android
+       --fast
+       --android-ndk-home=${ANDROID_NDK_PATH}
+       --force
+       --root="${SDL3_SOURCE_DIR}"
+    COMMAND sh -c "cp ${sdl3_SOURCE_DIR}/build-android/SDL3-*.aar ${SDL_ANDROID_ARCHIVE}"
+    COMMENT "Generating SDL3.aar for android-project"
+  )
+
+  set(GRADLEW "${CMAKE_SOURCE_DIR}/android-project/gradlew")
+  if(CMAKE_BUILD_TYPE MATCHES "^Rel")
+    set(GRADLE_TASK "assembleRelease")
+  else()
+    set(GRADLE_TASK "assembleDebug")
+  endif()
+
+  add_custom_target(ANDROID_GRADLE_BUILD ALL
+    DEPENDS "${SDL_ANDROID_ARCHIVE}"
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/android-project"
+    COMMAND "${GRADLEW}" ${GRADLE_TASK} -PcmakeArgs="-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DISLE_USE_DX5=${ISLE_USE_DX5} -DISLE_BUILD_CONFIG=${ISLE_BUILD_CONFIG} -DENABLE_CLANG_TIDY=${ENABLE_CLANG_TIDY} -DISLE_WERROR=${ISLE_WERROR} -DISLE_DEBUG=${ISLE_DEBUG} -Werror=dev"
+    VERBATIM
+    COMMENT "Building Android APK with Gradle"
+  )
+  message(STATUS "SDL AAR")
+  return()
```

Comment:
The build procedure is a little perverted. You're actually doing cmake -> gradle -> cmake.
Don't infect CMake with gradle please :)

I'd just require users to https://github.com/libsdl-org/SDL_helloworld/blob/501e31c4d9deac135b63b7535c2220d7d71d4d92/.github/workflows/main.yml#L51-L52 before doing anything isle related.
This can be moved to a separate script, downloading a SDL3 release or building a SDL3 aar.

On ci, you can then just do:
- download, or build SDL3 aar
- run gradle

(and not do apt-get)